src/parser/flags.ts

```diff
@@ -51,14 +53,18 @@ export const integer = (opts: { min?: number; max?: number} & Partial<OptionFlag
   })()
 }
 
-export const directory = (opts: { exists?: boolean } & Partial<OptionFlag<string>> = {}): OptionFlag<string | undefined> => {
+export function directory(opts: { exists?: boolean } & Partial<OptionFlag<string>> & ({required: true} | { default: number })): OptionFlag<string>
+export function directory(opts: { exists?: boolean } & Partial<OptionFlag<string>>): OptionFlag<string | undefined>
+export function directory(opts: { exists?: boolean } & Partial<OptionFlag<string>> = {}): OptionFlag<string> | OptionFlag<string | undefined> {
```

Comment:
Why is the duplicated `OptionFlag<T>` return type needed here?

```suggestion
export function directory(opts: { exists?: boolean } & Partial<OptionFlag<string>> = {}): OptionFlag<string | undefined> {
```

Reply:
@RodEsp if you don't have it then the overloads don't sufficiently overlap since the function either returns an `OptionFlag<T>` or a `OptionFlag<T | undefined>`